### PR TITLE
Fix CI failures

### PR DESCRIPTION
**Pull request by jean-roland**
Following #389, CI is functional again on this branch and it revealed many small issues that are fixed in this PR, along with a merge of the main branch to bring the aforementioned changes.

There is still an issue regarding fragmentation that seems to involve zenoh. Edit: solved

### CODE REVIEW

File: examples/unix/c11/z_queryable.c [hunk 1]
```diff
@@ -109,12 +109,9 @@ int main(int argc, char **argv) {
...
 
-    printf("Enter 'q' to quit...\n");
-    char c = '\0';
-    while (c != 'q') {
-        fflush(stdin);
-        int ret = scanf("%c", &c);
-        (void)ret;  // Remove unused result warning
+    printf("Press CTRL-C to quit...\n");
+    while (1) {
+        sleep(1);
```
Comment (Mallets) [inline examples/unix/c11/z_queryable.c:114]: Use `z_sleep`?

File: examples/unix/c11/z_sub.c [hunk 2]
```diff
@@ -108,12 +115,12 @@ int main(int argc, char **argv) {
...
-    while (c != 'q') {
-        fflush(stdin);
-        int ret = scanf("%c", &c);
-        (void)ret;  // Remove unused result warning
+    printf("Press CTRL-C to quit...\n");
+    while (1) {
+        if ((n != 0) && (msg_nb >= n)) {
+            break;
+        }
+        sleep(1);
```
Comment (Mallets) [inline examples/unix/c11/z_sub.c:123]: Use `z_sleep`?

File: examples/unix/c11/z_sub_st.c [hunk 3]
```diff
@@ -37,7 +37,7 @@ int main(int argc, char **argv) {
     const char *mode = "client";
     char *clocator = NULL;
     char *llocator = NULL;
-    int n = -1;
+    int n = 2147483647;  // max int value by default
```
Comment (Mallets) [inline examples/unix/c11/z_sub_st.c:40]: It's safer to use `int n = ~0;` in such a way we always have the guarantee `n` is `max(int)` regardless the actual size of `int`.

File: examples/unix/c99/z_sub.c [hunk 4]
```diff
@@ -93,12 +93,9 @@ int main(int argc, char **argv) {
...
 
-    printf("Enter 'q' to quit...\n");
-    char c = '\0';
-    while (c != 'q') {
-        fflush(stdin);
-        int ret = scanf("%c", &c);
-        (void)ret;  // Clear unused result warning
+    printf("Press CTRL-C to quit...\n");
+    while (1) {
+        sleep(1);
```
Comment (Mallets) [inline examples/unix/c99/z_sub.c:98]: Use `z_sleep`?

File: examples/unix/c99/z_queryable.c [hunk 5]
```diff
@@ -105,12 +105,9 @@ int main(int argc, char **argv) {
...
 
-    printf("Enter 'q' to quit...\n");
-    char c = '\0';
-    while (c != 'q') {
-        fflush(stdin);
-        int ret = scanf("%c", &c);
-        (void)ret;  // Clear unused result warning
+    printf("Press CTRL-C to quit...\n");
+    while (1) {
+        sleep(1);
```
Comment (Mallets) [inline examples/unix/c99/z_queryable.c:110]: Use `z_sleep`?

File: examples/freertos_plus_tcp/z_pub_st.c [hunk 6]
```diff
@@ -30,6 +30,7 @@
 
 #define KEYEXPR "demo/example/zenoh-pico-pub"
 #define VALUE "[FreeRTOS-Plus-TCP] Pub from Zenoh-Pico!"
+#define N 2147483647  // max int value by default
```
Comment (Mallets) [inline examples/freertos_plus_tcp/z_pub_st.c:33]: It's safer to use `int n = ~0;` in such a way we always have the guarantee `n` is `max(int)` regardless the actual size of `int`.

File: examples/freertos_plus_tcp/z_sub_st.c [hunk 7]
```diff
@@ -27,13 +27,17 @@
 #endif
 
 #define KEYEXPR "demo/example/**"
+#define N 2147483647  // max int value by default
```
Comment (Mallets) [inline examples/freertos_plus_tcp/z_sub_st.c:30]: It's safer to use `int n = ~0;` in such a way we always have the guarantee `n` is `max(int)` regardless the actual size of `int`.

File: examples/unix/c11/z_pub.c [hunk 8]
```diff
@@ -30,7 +30,7 @@ int main(int argc, char **argv) {
     const char *mode = "client";
     char *clocator = NULL;
     char *llocator = NULL;
-    int n = 10;
+    int n = 2147483647;  // max int value by default
```
Comment (Mallets) [inline examples/unix/c11/z_pub.c:33]: It's safer to use `int n = ~0;` in such a way we always have the guarantee `n` is `max(int)` regardless the actual size of `int`.

File: examples/unix/c11/z_pub_st.c [hunk 9]
```diff
@@ -27,7 +27,7 @@ int main(int argc, char **argv) {
     const char *mode = "client";
     char *clocator = NULL;
     char *llocator = NULL;
-    int n = 10;
+    int n = 2147483647;  // max int value by default
```
Comment (Mallets) [inline examples/unix/c11/z_pub_st.c:30]: It's safer to use `int n = ~0;` in such a way we always have the guarantee `n` is `max(int)` regardless the actual size of `int`.

File: examples/unix/c99/z_pub.c [hunk 10]
```diff
@@ -26,7 +26,7 @@ int main(int argc, char **argv) {
     const char *mode = "client";
     char *clocator = NULL;
     char *llocator = NULL;
-    int n = 10;
+    int n = 2147483647;  // max int value by default
```
Comment (Mallets) [inline examples/unix/c99/z_pub.c:29]: It's safer to use `int n = ~0;` in such a way we always have the guarantee `n` is `max(int)` regardless the actual size of `int`.

File: examples/unix/c99/z_pub_st.c [hunk 11]
```diff
@@ -26,9 +26,10 @@ int main(int argc, char **argv) {
     const char *mode = "client";
     char *clocator = NULL;
     char *llocator = NULL;
+    int n = 2147483647;  // max int value by default
```
Comment (Mallets) [inline examples/unix/c99/z_pub_st.c:29]: It's safer to use `int n = ~0;` in such a way we always have the guarantee `n` is `max(int)` regardless the actual size of `int`.

File: examples/windows/z_pub_st.c [hunk 12]
```diff
@@ -18,6 +18,8 @@
 #include <stdlib.h>
 #include <zenoh-pico.h>
 
+#define N 2147483647  // max int value by default
```
Comment (Mallets) [inline examples/windows/z_pub_st.c:21]: It's safer to use `int n = ~0;` in such a way we always have the guarantee `n` is `max(int)` regardless the actual size of `int`.

File: examples/windows/z_sub.c [hunk 13]
```diff
@@ -62,11 +62,9 @@ int main(int argc, char **argv) {
...
     }
 
-    printf("Enter 'q' to quit...\n");
-    char c = '\0';
-    while (c != 'q') {
-        fflush(stdin);
-        scanf("%c", &c);
+    printf("Press CTRL-C to quit...\n");
+    while (1) {
+        Sleep(1);
```
Comment (Mallets) [inline examples/windows/z_sub.c:67]: Use `z_sleep`?

File: examples/windows/z_sub_st.c [hunk 14]
```diff
@@ -18,13 +18,18 @@
 #include <stdlib.h>
 #include <zenoh-pico.h>
 
+#define N 2147483647  // max int value by default
```
Comment (Mallets) [inline examples/windows/z_sub_st.c:21]: It's safer to use `int n = ~0;` in such a way we always have the guarantee `n` is `max(int)` regardless the actual size of `int`.

File: examples/zephyr/z_queryable.c [hunk 15]
```diff
@@ -75,7 +75,7 @@ int main(int argc, char **argv) {
     printf("Zenoh setup finished!\n");
 
     while (1) {
-        sleep(5);
+        sleep(1);
```
Comment (Mallets) [inline examples/zephyr/z_queryable.c:78]: Use `z_sleep`?

File: examples/zephyr/z_sub.c [hunk 16]
```diff
@@ -69,7 +69,7 @@ int main(int argc, char **argv) {
     printf("OK!\n");
 
     while (1) {
-        sleep(5);
+        sleep(1);
```
Comment (Mallets) [inline examples/zephyr/z_sub.c:72]: Use `z_sleep`?

File: examples/espidf/z_queryable.c [hunk 17]
```diff
@@ -159,7 +159,7 @@ void app_main() {
     printf("Zenoh setup finished!\n");
 
     while (1) {
-        sleep(5);
+        sleep(1);
```
Comment (Mallets) [inline examples/espidf/z_queryable.c:162]: Use `z_sleep`?

File: examples/unix/c11/z_queryable.c (hunk 1 above)
Comment (jean-roland) [reply examples/unix/c11/z_queryable.c:114]: So this was imported from #359, I agree there should be a PR to normalize the examples but this is clearly outside of this PR scope.

File: src/system/espidf/network.c [hunk 18]
```diff
@@ -622,65 +624,74 @@ int8_t _z_listen_serial_from_dev(_z_sys_net_socket_t *sock, char *dev, uint32_t
     return ret;
 }
 
-void _z_close_serial(_z_sys_net_socket_t *sock) { uart_driver_delete(sock->_serial); }
+void _z_close_serial(_z_sys_net_socket_t *sock) {
+    uart_wait_tx_done(sock->_serial, 1000);
+    uart_flush(sock->_serial);
+    uart_driver_delete(sock->_serial);
+    zp_free(sock->after_cobs);
```
Comment (Mallets) [inline src/system/espidf/network.c:631]: Shouldn't be `z_free`?

File: src/system/espidf/network.c [hunk 19]
```diff
@@ -622,65 +624,74 @@ int8_t _z_listen_serial_from_dev(_z_sys_net_socket_t *sock, char *dev, uint32_t
...
+    zp_free(sock->before_cobs);
```
Comment (Mallets) [inline src/system/espidf/network.c:632]: Shouldn't be `z_free`?

File: examples/unix/c11/z_queryable.c (hunk 1 above)
Comment (Mallets) [reply examples/unix/c11/z_queryable.c:114]: Ok, noted. All other similar comments can be ignored then for this PR.

**Review by Mallets — APPROVED** [review]

**Review by Mallets — CHANGES_REQUESTED** [review]
Some minor changes, everything else LGTM.

File: examples/freertos_plus_tcp/z_pub_st.c (hunk 6 above)
Comment (Mallets) [reply examples/freertos_plus_tcp/z_pub_st.c:33]: You are right. Technically it should be an `unsigned int` and not an `int`.

**Comment by jean-roland** [comment]
We noticed some issues with examples:
* Some still use `sleep` instead of `z_sleep`
* We should use an unsigned int set to `~0` for the default loop count
But this is not the scope of this PR so this shall be adressed by another PR

**Comment by Mallets** [comment]
> We noticed some issues with examples:
> 
> * Some still use `sleep` instead of `z_sleep`
> * We should use an unsigned int set to `~0` for the default loop count
>   But this is not the scope of this PR so this shall be adressed by another PR

Agree

**Review by Mallets — APPROVED** [review]

**Pull request merged**